scripts/release.sh

```diff
@@ -285,7 +285,7 @@ if [ -n "$BUILD_BINARIES" ]; then
 fi
 
 for name in $IMAGES; do
-  image_basename=$($NIX_EVAL -f . images.$BUILD_TYPE.$name.imageName | xargs)
+  image_basename=$($NIX_EVAL -f . images.$BUILD_TYPE.$name.imageName --argstr product_prefix "$PRODUCT_PREFIX" | xargs)
```

Comment:
This seems unbound variable access?
Also if env var not set maybe we should not append --argstr at all

Reply:
We use the `set -u` option on the script. It's not going to be simple to add behaviour to a 'env variable not set' clause.

Reply:
Did you test this? I suspect this script will fail as it stands due to the -u...

You can do something like this:
```
if ! [ -z ${PRODUCT_PREFIX+x} ]; then
  PRODUCT_PREFIX=${PRODUCT_PREFIX:-"\"\""}
  PRODUCT_PREFIX="--argstr thing $PRODUCT_PREFIX"
else
  PRODUCT_PREFIX=""
fi
```

Reply:
Yeah the typical -z $PRODUCT_PREFIX check doesn't work. I did test it.

Reply:
Yes I did test the nix-eval too.